mlflow/store/dbmodels/models.py

```diff
@@ -45,11 +45,12 @@ def _create_entity(base, model):
                     metrics = {}
                     for o in obj:
                         existing_metric = metrics.get(o.key)
-                        if (existing_metric is None) or (o.timestamp > existing_metric.timestamp)\
-                            or (o.timestamp == existing_metric.timestamp
-                                and o.value > existing_metric.value):
-                            metrics[o.key] = Metric(o.key, o.value, o.timestamp)
-                    obj = metrics.values()
+                        if (existing_metric is None)\
+                            or ((o.step, o.timestamp, o.value) >=
+                                (existing_metric.step, existing_metric.timestamp,
+                                 existing_metric.value)):
+                            metrics[o.key] = Metric(o.key, o.value, o.timestamp, o.step)
```

Comment:
Do we have a test which confirms that we return the maximal metric which tests along the (step, timestamp, value) ordering?